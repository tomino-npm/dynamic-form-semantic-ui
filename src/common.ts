import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type SignatureType = {
  comment: string;
  signature: string;
  verifiedState: 'Pending' | 'Verified' | 'Rejected';
  rejected?: boolean;
  uid: string;
  name: string;
  date: Date;
  setValue?(name: string, value: any): any;
};

export type SignatureHandlers = {
  validateForm(): boolean;
  verifySignature(uid: string, signature: string): Promise<string>;
  sign(
    source: string,
    reject: boolean,
    password: string,
    reason: string,
    submit: boolean
  ): Promise<SignatureType>;
  signatureFont(): string;
};

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: any };
  readOnly: boolean;
};

export const { css, renderCss } = initCss('form-');
