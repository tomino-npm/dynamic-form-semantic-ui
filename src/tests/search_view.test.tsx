import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

const schema: JSONSchema = {
  type: 'object',
  properties: {
    country: {
      type: 'string'
    },
    selectedCountry: {
      type: 'string'
    }
  }
};

const options = [{ title: 'Australia', value: 'AUS' }, { title: 'Slovakia', value: 'SVK' }];

const controlledOptions = [
  { titles: ['Australia', 'Sydney'], value: 'SYD' },
  { titles: ['Australia', 'Melbourne'], value: 'SYD' },
  { titles: ['New Zealand', 'Christchurch'], title: 'Kiwis', value: 'SYD' }
];

const formDefinition: FormDefinition = create.form({
  elements: [
    create.formElement({
      row: 0,
      column: 0,
      width: 6,
      handler: 'findCountry',
      control: 'Search',
      label: 'Country',
      source: 'country'
    }),
    create.formElement({
      row: 0,
      column: 6,
      width: 6,
      handler: 'findCountryWithControl',
      control: 'Search',
      label: 'Country With Control',
      source: 'country',
      renderer: `<table><tr><td width="100">{0}<!--country--></td><td>{1}<!--city--></td></tr></table>`
    }),
    create.formElement({
      row: 0,
      column: 12,
      width: 4,
      handler: 'findCountry',
      control: 'Search',
      label: 'Selected',
      source: 'selectedCountry'
    })
  ]
});

function componentWithData() {
  const form = new FormModel(formDefinition, schema, { selectedCountry: 'SVK' });

  // just another notation
  return (
    <TestComponent
      form={form}
      handlers={{
        search: (name: string, searchString: string, id: string) => {
          if (name === 'findCountry') {
            return new Promise(resolve => {
              setTimeout(() => {
                resolve(
                  options.filter(o => (id ? o.value === id : o.title.indexOf(searchString) >= 1))
                );
              }, 100);
            });
          } else {
            return new Promise(resolve => {
              setTimeout(() => {
                resolve(
                  controlledOptions.filter(o =>
                    id ? o.value === id : o.titles[0].indexOf(searchString) >= 1
                  )
                );
              }, 100);
            });
          }
        },
        findCountryWithControl: () => {
          return new Promise(resolve => {
            setTimeout(() => {
              resolve(controlledOptions);
            }, 100);
          });
        }
      }}
    />
  );
}

describe('Search', () => {
  it('renders', () => {
    const component = renderer.create(componentWithData());
    expect(component).toMatchSnapshot();
  });
  return { componentWithData };
});
