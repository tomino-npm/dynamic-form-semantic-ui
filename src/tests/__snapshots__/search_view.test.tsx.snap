// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<div
  className="ui segment ui form"
>
  <div
    className="ui dividing header"
  >
    Standard
  </div>
  <div
    className="ui form "
  >
    <div
      className="fields"
    >
      <div
        className="six wide field"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <div
          className="ui loading search"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
        >
          <div
            className="ui icon input"
          >
            <input
              autoComplete="off"
              className="prompt"
              onChange={[Function]}
              onClick={[Function]}
              tabIndex="0"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="six wide field"
      >
        <label
          htmlFor="country"
        >
          Country With Control
        </label>
        <div
          className="ui loading search"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
        >
          <div
            className="ui icon input"
          >
            <input
              autoComplete="off"
              className="prompt"
              onChange={[Function]}
              onClick={[Function]}
              tabIndex="0"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="search icon"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui dividing header"
  >
    Readonly
  </div>
  <div
    className="ui form form-hjcaghiia"
  >
    <div
      className="fields"
    >
      <div
        className="six wide field"
      >
        <label
          htmlFor="country"
        >
          Country
        </label>
        <div
          className="ui disabled input"
        >
          <input
            disabled={true}
            onChange={[Function]}
            tabIndex={-1}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="six wide field"
      >
        <label
          htmlFor="country"
        >
          Country With Control
        </label>
        <div
          className="ui disabled input"
        >
          <input
            disabled={true}
            onChange={[Function]}
            tabIndex={-1}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
